xtrack/beam_elements/elements.py

```diff
@@ -84,6 +84,171 @@ def add_slice(weight, container, thick_name, slice_name, _buffer=None):
+		domegax: float
+				A floating point number representing the value of the horizontal chromaticity 
+				in the ring multiplied by 2pi. Default is ``0``.
+		domegay: float
+				A floating point number representing the value of the vertical chromaticity 
+				in the ring multiplied by 2pi. Default is ``0``.
+		dx: float
+				A floating point number representing the value of horizontal dispersion at 
+				the location the multipole. Default is ``0``.
+		ddx: float
+				A floating point value representing the value of the derivative of horizontal 
+				dispersion at the location of the multipole. Default is ``0``.
+    fx_coeffs: array of floats
+        An array that contains the coefficients of monomials of the form x^n*y*m that 
+        represent the nonlinearities of the map in the horizontal plane. It is 
+        calculated at initialisation based on the multipole coefficients provided.
+    fx_x_exps: array of floats
+        An array containing the exponents, n, of x for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the horizontal plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fx_y_exps: array of floats
+        An array containing the exponents, m, of y for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the horizontal plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fy_coeffs: array of floats
+        An array that contains the coefficients of monomials of the form x^n*y*m that 
+        represent the nonlinearities of the map in the vertical plane. It is 
+        calculated at initialisation based on the multipole coefficients provided.
+    fy_x_exps: array of floats
+        An array containing the exponents, n, of x for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the vertical plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    fy_y_exps: array of floats
+        An array containing the exponents, m, of y for all monomials of the form 
+        x^n*y^m that represent the nonlinearities of the map in the vertical plane. 
+        It is calculated at initialisation based on the multipole coefficients provided.
+    n_fx_coeffs: int
+        Length of the arrays fx_coeffs, fx_x_exps, and fx_y_exps.
+    n_fy_coeffs: int
+        Length of the arrays fy_coeffs, fy_x_exps, and fy_y_exps.
+    n_turns: int
+        Number of turns to track. Default is ``1``.
+    norm: int
+        1 if input coordinates are already normalised, 0 if not. Default is ``0``.
+
+    Comments
+    --------
+
+    - The user provides omega_x and omega_y, their sin and cos is calculated at 
+      initialisation.
+    - The user provides "multipole_coeffs", an array of floats, that contains the 
+      strength of the multipoles present in the map. Default is ``[0]``.
```

Comment:
I would put these in the parameters section. More generally, the parameters section should contain the input allowed from the user.

On a related note, you accept both `omega_x` and `sin_omega_x` (and analogously for cos and y). It could be worth it to add a consistency check, or only accept angle if and only if no trig values are given (the latter may work, since you don't store the angle, otherwise it would be a problem when loading the values back from dict; for the former you can look at https://github.com/xsuite/xtrack/blob/main/xtrack/beam_elements/elements.py#L424 and use the helper function.) Likely the same comment applies for `multipole_coeffs`.

Reply:
Sorry, I thought the parameters should be the _xofields attributes, so I updated the comment to contain under "Parameters" the user input. However, I did write the element such that it runs as fast as possible, i.e. avoiding any unnecessary trig evaluations etc. For this reason I don't store the `omega_x`, only its sine and cosine. Thus I also don't accept e.g. `sin_omega_x` as input. The user can still change the tune during a many-turn simulation for example, but they must then provide `sin_omega_x` and `cos_omega_x` directly (i.e. `henon.sin_omega_x = my_new_value`). Likewise, they can change multipole strength, but they must provide the `fx_x_coeffs`, `fy_x_coeffs` etc. instead of simply the integrated multipole strength they provided at initialization. This is not a super likely usecase, but I myself do this sometimes, so I thought it important to point this out, and I added a "Comment" section with a detailed explanation of the `_xofields` attributes and a "Notes" section that should make it clear for the user how they can change these properties of the map if they need to.